client/src/components/JurisdictionAdmin/generateSheets.ts

```diff
@@ -200,10 +202,354 @@ export const downloadAuditBoardCredentials = async (
       auditBoardCreds.text(`${name}: No ballots`, 20, i * 10 + 20)
     })
   }
-  auditBoardCreds.autoPrint()
   await auditBoardCreds.save(
     `Audit Board Credentials - ${jurisdictionName} - ${auditName}.pdf`,
     { returnPromise: true }
   )
   return auditBoardCreds.output() // returned for test snapshots
 }
+
+export const downloadBatchTallySheets = async (
+  batches: IBatch[],
+  contestChoices: ICandidate[],
+  jurisdictionName: string
+): Promise<string> => {
+  const doc = new jsPDF({ format: 'letter', unit: 'pt' })
+
+  const pageHeight = 792 // 11 inches * 72 pts per inch
+  const pageWidth = 612 // 8.5 inches
+  const pageMargin = 72 // 1 inch
+  const pageContentWidth = pageWidth - pageMargin * 2
+
+  const defaultFontSize = 12
+  const headingFontSize = 18
+  const subHeadingFontSize = 16
+  const sectionBottomMargin = 24
+  const pBottomMargin = 10
+  const drawingLineWidth = 1
+
+  const checkboxSize = 10
+  const checkboxLeftMargin = 10
+  const checkboxRightMargin = 6
+  const checkboxTopMargin = 1 // To properly align checkboxes with text
+
+  const tableCellMinWidth = 100
+  const tableCellPadding = 6
+
+  const signatureLineLabelFontSize = 10
+  const signatureLineRightMargin = 10
+  const signatureLineLabelTopMargin = 6
+
+  let y = pageMargin
+  const yMax = pageHeight - pageMargin
+  for (let i = 0; i < batches.length; i += 1) {
+    const batch = batches[i]
+
+    doc.setFont('Helvetica', 'normal').setFontSize(headingFontSize)
+    doc.setLineWidth(drawingLineWidth).setDrawColor('black')
+
+    doc.text('Audit Board Batch Tally Sheet', pageMargin, y)
+    y += doc.getLineHeight() + pBottomMargin
+
+    doc.setFont('Helvetica', 'bold').setFontSize(subHeadingFontSize)
+
+    y = renderTextWrapped({
+      doc,
+      text: `Batch Name: ${batch.name}`,
+      wrapWidth: pageContentWidth,
+      x: pageMargin,
+      y,
+      bottomMargin: sectionBottomMargin,
+    })
+
+    doc.setFont('Helvetica', 'normal').setFontSize(defaultFontSize)
+
+    y = renderTextWrapped({
+      doc,
+      text: `Jurisdiction: ${jurisdictionName}`,
+      wrapWidth: pageContentWidth,
+      x: pageMargin,
+      y,
+      bottomMargin: pBottomMargin,
+    })
+
+    y = renderTextWrapped({
+      doc,
+      text: `Audit Board: ${batch.auditBoard ? batch.auditBoard.name : ''}`,
```

Comment:
`batch.auditBoard` should always be defined at this point. I'm not sure if failing loudly or failing silently is better

Reply:
Hmm yeah, though I didn't know `batch.auditBoard` would always be defined here, I've encountered other cases like this and often feel on the fence about whether to fail loudly or silently. I can raise this as a topic for an eng discussion!